### ISSUE TITLE
Memory leak when initializing the connections with a lot of sites and a lot of languages

### ISSUE THREAD

**Issue by dogawaf** [TYPO3-Solr/ext-solr#1288]
Hello

I have a TYPO3 instance with 100+ rootpages and 27 languages.
When initializing the connection in the cache menu, php runs out of memory and/or end with a timeout.

I think that loading the TS setup 100*27 times is the cause :)

Initializing the connection site by site with the context menu item in the page tree is OK.

TYPO3 7.6
ext:solr 6.1

Regards.

**Comment by dogawaf** [comment]
Hello
I tried to initialize the connections with the given extbase cli command, with a lot of memory (2GB), but unfortunately the cli command crashed with some fatal error that I don't remember exactly: it was about a ContentObject, which is weird because of the CLI context.

**Comment by dogawaf** [comment]
I've uploaded a [simple log](https://github.com/TYPO3-Solr/ext-solr/files/1037939/solr_updateconnections.txt) of `php ./typo3/cli_dispatch.phpsh extbase solr:updateconnections`.
Each line show a rootpage|language --> memory usage.
I've put the echo in `\ApacheSolrForTypo3\Solr\ConnectionManager::getConfiguredSolrConnectionByRootPage()`.

The script does not end, the registry is not updated without any error.

I think it would be helpful to find a way to empty some TYPO3's memory cache at the end of getConfiguredSolrConnectionByRootPage, but I did not find a way to do that.

Anyway... in order to manage my "edge case", I would like to be able to execute the CLI command by passing a rootpage id, like that: 
```
php ./typo3/cli_dispatch.phpsh extbase solr:updateconnections --rootpageid 121
```
It's easy to do, I can make a pull request if you are OK with that.

Regards
